File: references/library-project/backend/src/main/scala/services/books/objects/BookRecord.scala
package services.books.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
import services.books.objects.TimeCodecs.given

import java.time.Instant

final case class BookRecord(
  id: BookId,
  title: String,
  author: String,
  isbn: String,
  category: String,
  categoryLabel: String,
  stockTotal: Int,
  stockAvailable: Int,
  summary: String,
  status: BookInventoryStatus,
  createdAt: Instant,
  updatedAt: Instant
)

object BookRecord:
  given Encoder[BookRecord] = deriveEncoder[BookRecord]
  given Decoder[BookRecord] = deriveDecoder[BookRecord]

File: references/library-project/frontend/src/objects/books/BookRecord.ts
import type { BookId } from './BookId'
import type { BookInventoryStatus } from './BookInventoryStatus'
import type { InstantString } from './TimeCodecs'

export interface BookRecord {
  id: BookId
  title: string
  author: string
  isbn: string
  category: string
  categoryLabel: string
  stockTotal: number
  stockAvailable: number
  summary: string
  status: BookInventoryStatus
  createdAt: InstantString
  updatedAt: InstantString
}
